bots/csr/src/main/java/org/openjdk/skara/bots/csr/CSRBot.java

```diff
@@ -72,6 +73,22 @@ public static Optional<JdkVersion> getVersion(PullRequest pullRequest) {
         return JdkVersion.parse(version);
     }
 
+    private boolean hasCsrIssueAndProgress(PullRequest pr, Issue csr) {
+        return hasCsrIssue(pr, csr) &&
+                (pr.body().contains("- [ ] Change requires a CSR request to be approved") ||
+                 pr.body().contains("- [x] Change requires a CSR request to be approved"));
+    }
+
+    private boolean hasCsrIssueAndProgressChecked(PullRequest pr, Issue csr) {
+        return hasCsrIssue(pr, csr) && pr.body().contains("- [x] Change requires a CSR request to be approved");
+    }
+
+    private boolean hasCsrIssue(PullRequest pr, Issue csr) {
+        return pr.body().contains(csr.id()) &&
+                pr.body().contains(csr.webUrl().toString()) &&
+                pr.body().contains(csr.title() + " (**CSR**)");
+    }
```

Comment:
We could consider making these checks safer by only looking in the Skara generated part of the body. There is a delimiter line you could split on.

Reply:
Fixed.